File: lab-redes/python/websocket/mural_servidor.py
import asyncio
import websockets

clientes_conectados = set()

async def tratar_conexao(websocket):
    clientes_conectados.add(websocket)
    print(f"[WebSocket] Novo aluno conectado. Total: {len(clientes_conectados)}")
    await websocket.send("Bem-vindo(a) ao mural de avisos da turma!")
    try:
        async for mensagem in websocket:
            print(f"[WebSocket] Recebido: {mensagem}")
            aviso_formatado = f"Aviso da turma: {mensagem}"
            websockets.broadcast(clientes_conectados, aviso_formatado)
    finally:
        clientes_conectados.remove(websocket)
        print(f"[WebSocket] Aluno desconectado. Total: {len(clientes_conectados)}")

async def main():
    # Rodando em máquina compartilhada com colegas? Some seu OFFSET (seção 3.3): 8888 + OFFSET
    print("[WebSocket] Servidor do mural iniciado na porta 8888.")
    async with websockets.serve(tratar_conexao, "0.0.0.0", 8888):
        await asyncio.Future()  # mantém o servidor rodando indefinidamente

asyncio.run(main())

File: lab-redes/java/websocket/src/main/java/MuralServidor.java
import org.java_websocket.WebSocket;
import org.java_websocket.handshake.ClientHandshake;
import org.java_websocket.server.WebSocketServer;

import java.net.InetSocketAddress;

public class MuralServidor extends WebSocketServer {

    public MuralServidor(int porta) {
        super(new InetSocketAddress(porta));
    }

    @Override
    public void onOpen(WebSocket conexao, ClientHandshake handshake) {
        System.out.println("[WebSocket] Novo aluno conectado: " + conexao.getRemoteSocketAddress());
        conexao.send("Bem-vindo(a) ao mural de avisos da turma!");
    }

    @Override
    public void onMessage(WebSocket conexao, String mensagem) {
        System.out.println("[WebSocket] Recebido: " + mensagem);
        String avisoFormatado = "Aviso da turma: " + mensagem;
        for (WebSocket cliente : getConnections()) {
            cliente.send(avisoFormatado);
        }
    }

    @Override
    public void onClose(WebSocket conexao, int codigo, String motivo, boolean remoto) {
        System.out.println("[WebSocket] Aluno desconectado: " + conexao.getRemoteSocketAddress());
    }

    @Override
    public void onError(WebSocket conexao, Exception ex) {
        ex.printStackTrace();
    }

    @Override
    public void onStart() {
        System.out.println("[WebSocket] Servidor do mural iniciado.");
    }

    public static void main(String[] args) {
        // Rodando em máquina compartilhada com colegas? Some seu OFFSET (seção 3.3): 8887 + OFFSET
        MuralServidor servidor = new MuralServidor(8887);
        servidor.start();
    }
}